go-controller/pkg/cni/helper_linux.go

```diff
@@ -441,26 +449,35 @@ func (pr *PodRequest) ConfigureInterface(podLister corev1listers.PodLister, kcli
 	}
 	defer netns.Close()
 
+	pod, err := kclient.CoreV1().Pods(pr.PodNamespace).Get(pr.ctx, pr.PodName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
 	var hostIface, contIface *current.Interface
 
 	klog.V(5).Infof("CNI Conf %v", pr.CNIConf)
 	if pr.CNIConf.DeviceID != "" {
 		// SR-IOV Case
-		hostIface, contIface, err = setupSriovInterface(netns, pr.SandboxID, pr.IfName, ifInfo, pr.CNIConf.DeviceID)
+		hostIface, contIface, err = setupSriovInterface(netns, pr.SandboxID, pr.IfName, ifInfo, pr.CNIConf.DeviceID, pod)
 	} else {
 		if ifInfo.IsDPUHostMode {
 			return nil, fmt.Errorf("unexpected configuration, pod request on dpu host. " +
 				"device ID must be provided")
 		}
 		// General case
-		hostIface, contIface, err = setupInterface(netns, pr.SandboxID, pr.IfName, ifInfo)
+		hostIface, contIface, err = setupInterface(netns, pr.SandboxID, pr.IfName, ifInfo, pod)
```

Comment:
it looks like this setups up the veth pair. In the future, is it possible for us to create a tap or macvtap device here instead so we can avoid the bridge binding when passing the interface to qemu-kvm?

Reply:
Is tricky business, the poc does not need it.